### PR TITLE
Check-health for Kafka Collector

### LINKED ISSUE #2098: Make kafka health check meaningful
Right now, the kafka health check returns ok eventhough it is not. Let's improve that.

### PR DESCRIPTION
**Pull request by jeqo**
Use Admin Client API to validate that communication with Kafka brokers is working correctly.

Attempt to fix #2098

### CODE REVIEW

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java [hunk 1]
```diff
@@ -148,9 +153,12 @@ public KafkaCollector build() {
   }
 
   final LazyKafkaWorkers kafkaWorkers;
+  private final Properties properties;
+  private AdminClient adminClient;
```
Comment (zeagord) [inline zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: I would change this to package specific access to reduce the size of the bytecode.

**Review by zeagord — APPROVED** [review]

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java (hunk 1 above)
Comment (jeqo) [reply zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: @zeagord didn't know that, thanks! :) Fixing it.

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java [hunk 2]
```diff
@@ -164,15 +172,19 @@ public CheckResult check() {
     try {
       CheckResult failure = kafkaWorkers.failure.get(); // check the kafka workers didn't quit
       if (failure != null) return failure;
+      if (adminClient == null) adminClient = AdminClient.create(properties);
+      KafkaFuture<String> maybeClusterId = adminClient.describeCluster().clusterId();
+      maybeClusterId.get(1, TimeUnit.SECONDS);
       return CheckResult.OK;
-    } catch (RuntimeException e) {
+    } catch (RuntimeException | TimeoutException | ExecutionException | InterruptedException e) {
```
Comment (codefromthecrypt) [inline zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: probably easier to just catch exception

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java [hunk 3]
```diff
@@ -164,15 +172,19 @@ public CheckResult check() {
     try {
       CheckResult failure = kafkaWorkers.failure.get(); // check the kafka workers didn't quit
       if (failure != null) return failure;
+      if (adminClient == null) adminClient = AdminClient.create(properties);
```
Comment (codefromthecrypt) [inline zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: if the goal is to initialize (then later close) the admin client once, I think you will want to double check initialize with volatile ref like some other code in this repo

reason is close isnt guaranteed to be same thread as check

**Review by codefromthecrypt — COMMENTED** [review]
small note otherwise lgtm thanks!

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java (hunk 2 above)
Comment (jeqo) [reply zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: Agree. I'll fix it.

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java (hunk 3 above)
Comment (jeqo) [reply zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java]: Thanks for the advise, I'll check it.

**Comment by codefromthecrypt** [comment]
lgtm!

**Comment by codefromthecrypt** [comment]
can you do the same in zipkin-reporter?

**Pull request merged**